apps/cmp.c

```diff
@@ -1918,15 +1918,19 @@ static int setup_client_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
                 goto err;
             }
         }
+
+        /* Will no more be needed after fixing callback design flaw #17088 */
         if ((info = OPENSSL_zalloc(sizeof(*info))) == NULL)
             goto err;
         (void)OSSL_CMP_CTX_set_http_cb_arg(ctx, info);
         /* info will be freed along with CMP ctx */
         info->server = opt_server;
         info->port = server_port;
-        info->use_proxy = opt_proxy != NULL;
+        info->use_proxy = /* workaround for callback design flaw, see #17088 */
```

Comment:
don't format that way - comment on line before ... don't use a comment to break a single line of code

Reply:
> 

This surprises me, for several reasons:
* I've been using comments within (longer) C statements for long time at various places without getting feedback on this.
* The official - rather dusted - coding style at https://www.openssl.org/policies/codingstyle.html does not prevent/discourage this. 
* The update that I pushed on at #10725, which BTW has not enjoyed attention for some 8 months, does not include such a requirement, neither. Consequently, also `check-forma.pl` does not check for it.

Reply:
How to proceed here?
Are intra-statement comments considered acceptable or not?

Reply:
Considering that this comment is about the whole statement, it makes no sense to have it in the middle like this.

Personally, though, I would avoid referring to github issues and PR in code.  It's better to explain in the comment what's going on.

Reply:
> Considering that this comment is about the whole statement, it makes no sense to have it in the middle like this.

Having a comment like this on the same line as the code it refers to has the advantage that it is crystal clear to which statement it belongs.
Yet given the convention that usually a comment refers to the following line(s), it's ok for me to move it to a separate line above the statement - done so.

> Personally, though, I would avoid referring to github issues and PR in code. It's better to explain in the comment what's going on.

I generally agree, but explaining in this case what's going on would become pretty lengthy.

Anyway, the comment will go away when #17088 can be merged for the next major release.